includes/api/class-wc-admin-rest-leaderboards-controller.php

```diff
@@ -0,0 +1,481 @@
+				),
+			),
+			'rows'    => $rows,
+		);
+	}
+
+	/**
+	 * Get the data for the products leaderboard.
+	 *
+	 * @param int    $rows Number of rows.
+	 * @param string $after Items after date.
+	 * @param string $before Items before date.
+	 * @param string $persisted_query URL query string.
+	 */
+	public function get_products_leaderboard( $rows, $after, $before, $persisted_query ) {
+		$products_data_store = new WC_Admin_Reports_Products_Data_Store();
+		$products_data       = $products_data_store->get_data(
+			array(
+				'orderby'       => 'items_sold',
+				'order'         => 'desc',
+				'after'         => $after,
+				'before'        => $before,
+				'extended_info' => true,
+			)
+		);
+
+		$row = array();
+		foreach ( $products_data->data as $product ) {
+			$url_query    = wp_parse_args(
+				array(
+					'filter'   => 'single_product',
+					'products' => $product['product_id'],
+				),
+				$persisted_query
+			);
+			$product_url  = wc_admin_url( 'analytics/products', $url_query );
+			$product_name = $product['extended_info'] && $product['extended_info']['name'] ? $product['extended_info']['name'] : '';
+			$rows[]       = array(
+				array(
+					'display' => "<a href='{$product_url}'>{$product_name}</a>",
+					'value'   => $product_name,
+				),
+				array(
+					'display' => wc_admin_number_format( $product['items_sold'] ),
+					'value'   => $product['items_sold'],
+				),
+				array(
+					'display' => wc_price( $product['net_revenue'] ),
+					'value'   => $product['net_revenue'],
+				),
+			);
+		}
+
+		return array(
+			'id'      => 'products',
+			'label'   => __( 'Top Products - Items Sold', 'woocommerce-admin' ),
+			'headers' => array(
+				array(
+					'label' => __( 'Product', 'woocommerce-admin' ),
+				),
+				array(
+					'label' => __( 'Items Sold', 'woocommerce-admin' ),
+				),
+				array(
+					'label' => __( 'Net Revenue', 'woocommerce-admin' ),
+				),
+			),
+			'rows'    => $rows,
+		);
+	}
+
+	/**
+	 * Get an array of all leaderboards.
+	 *
+	 * @param int    $rows Number of rows.
+	 * @param string $after Items after date.
+	 * @param string $before Items before date.
+	 * @param string $persisted_query URL query string.
+	 * @return array
+	 */
+	public function get_leaderboards( $rows, $after, $before, $persisted_query ) {
+		$leaderboards = array(
+			$this->get_customers_leaderboard( $rows, $after, $before, $persisted_query ),
+			$this->get_coupons_leaderboard( $rows, $after, $before, $persisted_query ),
+			$this->get_categories_leaderboard( $rows, $after, $before, $persisted_query ),
+			$this->get_products_leaderboard( $rows, $after, $before, $persisted_query ),
+		);
+
+		return apply_filters( 'woocommerce_leaderboards', $leaderboards, $rows, $after, $before, $persisted_query );
+	}
+
+	/**
+	 * Return all leaderboards.
+	 *
+	 * @param  WP_REST_Request $request Request data.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		global $wpdb;
```

Comment:
This looks to be unused.

Reply:
Good eyes- this is no longer being used.